Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +11,337 @@ public static class BalanceStats
+                decimal.TryParse(value, out decimal currentBalance);
+
+                if (max < currentBalance)
+                {
+                    max = currentBalance;
+                    richestPeople.Clear();
+                    richestPeople.Append(auxBalances[0]);
+                    moreThanOne = false;
+                    result = $" is the richest person. {max:C0}.";
+                }
+                else if(max.Equals(currentBalance))
+                {
+                    richestPeople.Append($", {auxBalances[0]}");
+                    result = $" are the richest people. {max:C0}.";
+                    moreThanOne = true;
+                }
+            }
+
+            if (moreThanOne)
+            {
+                ReplaceLastCommaWithAnd(richestPeople);
+            }
+            richestPeople.Append(result);
+            return richestPeople.ToString();
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal min = decimal.MaxValue;
+            string result = "";
+            StringBuilder poorPeople = new StringBuilder();
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                string value = auxBalances[^1..][0];
+                decimal.TryParse(value, out decimal currentBalance);
+                if (min > currentBalance)
+                {
+                    min = currentBalance;
+                    poorPeople.Clear();
+                    poorPeople.Append(auxBalances[0]);
+                    result = $" has the least money. {min:C0}.";
+                    moreThanOne = false;
+                }
+                else if(min.Equals(currentBalance))
+                {
+                    poorPeople.Append($", {auxBalances[0]}");
+                    result = $" have the least money. {min:C0}.";
+                    moreThanOne = true;
+                }
+            }
+
+            if (moreThanOne)
+            {
+                ReplaceLastCommaWithAnd(poorPeople);
+            }
+            poorPeople.Append(result);
+            return poorPeople.ToString();
+        }
+
+        /// <summary>
+        /// Makes a customizable invariant culture without currencygroupseparator and currencynegativepattern = 1
+        /// </summary>
+        /// <returns>an Invariant Culture CultureInfo</returns>
+        private static CultureInfo GetCustomInvariantCulture()
+        {
+            CultureInfo cult = (CultureInfo)CultureInfo.InvariantCulture.Clone();
```

Comment:
Just call it full name- culture. cult gives it a whole different funny meaning.

Reply:
that was my intention ;-)